velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -716,6 +716,57 @@ PlanBuilder& PlanBuilder::localMerge(const std::vector<std::string>& keys) {
   return *this;
 }
 
+PlanBuilder& PlanBuilder::expand(
+    const std::vector<std::vector<std::string>>& projectionSets) {
+  std::vector<std::vector<core::TypedExprPtr>> projectSetExprs;
+  projectSetExprs.reserve(projectionSets.size());
+  std::vector<std::string> names;
+  names.reserve(projectionSets[0].size());
+  std::vector<std::shared_ptr<const Type>> types;
+  types.reserve(projectionSets[0].size());
+  std::string groupIdPrefix = "group_id_";
+  int grouIdColCount = 0;
+  for (int i = 0; i < projectionSets[0].size(); ++i) {
+    for (int j = 0; j < projectionSets.size(); ++j) {
+      if (projectionSets[j][i] != "") {
+        if (planNode_->outputType()->containsChild(projectionSets[j][i])) {
```

Comment:
In other PlanBuilder methods, strings usually represent SQL expressions. Here, it seems that 'projections' represent input column names. Is this so? This needs to be clarified in the .h file. Also, how would a user create a constant null or non-null projection? These are supported by the Expand operator, but looks like not supported in the PlanBuilder. Please, reconsider.

Reply:
@JkSelf I'm not sure I fully understand. Would you show how to use PlanBuilder to create each type of supported projection?

Reply:
@mbasmanova  Assuming the input data type is k1, k2, a, b, and the expand projections are as follows:
```
{"k1", "", "a", "b", "0", "0"},
{"k1", "", "a", "b", "0", "1"},
{"", "k2", "a", "b", "1", "2"}
```

Each item in these projections can represent a column name, be empty, or be a constant value. Constant columns are not part of the input row type. They are newly generated columns by the Expand operator. For example, the output of the above projections includes five columns: k1, k2, a, b, group_id_0, group_id_1. During the PlanBuild process, we iterate through two layers of for loops based on the projections to generate the name and type of each column.
Considering the given projections:
- For the first column, we traverse the projections and find the first non-empty element, which is projections[0][0]. Then, we break out of the inner for loop.
- For the second column, we find the non-empty element as projections[2][1].
- The third column corresponds to projections[0][2], and the fourth column corresponds to projections[0][3].
- Columns 5 and 6 are constant columns that are not part of the input row type.
- Consequently, we create new columns named group_id_0 and group_id_1.